internal/provider/provider.go

```diff
@@ -228,6 +228,19 @@ func azureProvider(supportLegacyTestSuite bool) *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("ARM_STORAGE_USE_AZUREAD", false),
 				Description: "Should the AzureRM Provider use AzureAD to access the Storage Data Plane API's?",
 			},
+
+			"use_msal": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Should Terraform obtain MSAL auth tokens and no longer use Azure Active Directory Graph? Defaults to false, unless the 3.0 beta is enabled in which case defaults to true",
+				DefaultFunc: func() schema.SchemaDefaultFunc {
+					defaultVal := false
+					if features.ThreePointOh() {
+						defaultVal = true
```

Comment:
for now we can return true here? I don't think we need to support ADAL through 4.0 tbh?

```suggestion
						return true
```

Reply:
OK sounds good. I was going for deprecation in v3 but makes sense to move a bit quicker since this _should_ be a non-user-facing change :)